lib/utils.js

```diff
@@ -60,6 +66,34 @@ function getAddressInfo(feature) {
   return addrInfo;
 }
 
+/**
+ * This function transforms the v6 feature from reverse geocoding to AddressInfo object
+ * @param {object} feature 
+ * @returns {object}
+ */
+function getV6AddressInfo(feature) {
+  const houseNumber = feature.address_number || "";
+  const street = feature.street || "";
+  const placeName = feature.name + (feature.place_formatted ? `, ${feature.place_formatted}` : "");
+  const address = feature.name;
+
+  const addrInfo = {
+    address: address,
+    houseNumber: houseNumber,
+    street: street,
+    placeName: placeName,
+  };
+
+  for (const contextProp in feature.context) {
+    if (contextProp) {
+      const layer = contextProp.id.split(".")[0];
+      addrInfo[layer] = contextProp.name;
```

Comment:
I believe you have you index the `feature.context` object with the `contextProp` key:
```js
addrInfo[contextProp] = feature.context[contextProp].name
```
otherwise `contextProp` only refers to the key rather than the full object